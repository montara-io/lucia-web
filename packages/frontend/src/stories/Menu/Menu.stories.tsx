import React from 'react';
import Menu from './Menu';
import { action } from '@storybook/addon-actions';

export default {
  title: 'Components/Menu',
  component: Menu,
};

export const Basic = (args) => {
  return <Menu {...args} />;
};

Basic.args = {
  id: 'menu',
  profileImage: '/assets/avatar.png',
  model: {
    body: [
      {
        label: '',
        items: [
          {
            id: 'dashboard',
            label: 'DASHBOARD',
            icon: '/assets/icons/home.svg',
            command: action('clicked'),
          },
        ],
      },
      {
        label: 'TEAM',
        items: [
          {
            id: 'team-engagement',
            label: 'ENGAGEMENT',
            icon: '/assets/icons/pulse.svg',
            command: action('clicked'),
          },
          {
            id: 'team-productivity',
            label: 'PRODUCTIVITY',
            icon: '/assets/icons/energy.svg',
            command: action('clicked'),
          },
          {
            id: 'team-wellbeing',
            label: 'WELLBEING',
            icon: '/assets/icons/calm.svg',
            command: action('clicked'),
            disabled: true,
          },
          {
            id: 'team-team-members',
            label: 'TEAM MEMBERS',
            icon: '/assets/icons/team.svg',
            command: action('clicked'),
            disabled: true,
          },
        ],
      },
      {
        label: 'ME',
        items: [
          {
            id: 'productivity',
            label: 'PRODUCTIVITY',
            icon: '/assets/icons/energy.svg',
            command: action('clicked'),
          },
          {
            id: 'wellbeing',
            label: 'WELLBEING',
            icon: '/assets/icons/calm.svg',
            command: action('clicked'),
          },
          {
            id: 'leadership',
            label: 'LEADERSHIP',
            icon: '/assets/icons/goal.svg',
            command: action('clicked'),
          },
          {
            id: 'career',
            label: 'CAREER GROWTH',
            icon: '/assets/icons/compass.svg',
            command: action('clicked'),
          },
        ],
      },
    ],
    footer: [
      {
        label: '',
        items: [
          {
            id: 'settings',
            label: 'SETTINGS',
            icon: '/assets/icons/settings.svg',
            command: action('clicked'),
          },
          {
            id: 'logout',
            label: 'LOGOUT',
            icon: '/assets/icons/log-out.svg',
          },
        ],
      },
    ],
  },
  onSelected: action('Selected'),
  onExpanded: action('Expanded'),
  defaultSelected: 'dashboard',
};

export const SlimManager = (args) => {
  return <Menu {...args} />;
};

SlimManager.args = {
  id: 'menu',
  profileImage: '/assets/avatar.png',
  model: {
    body: [
      {
        label: '',
        items: [
          {
            id: 'dashboard',
            label: 'DASHBOARD',
            icon: '/assets/icons/home.svg',
            command: action('clicked'),
          },
        ],
      },
      {
        label: 'TEAM',
        disabled: true,
        tooltip: {
          title: 'Team Safe Zone',
          content:
            'Montara respects your people’s privacy. These trackers are available for teams with at least 5 people.',
        },
        items: [
          {
            id: 'team-engagement',
            label: 'ENGAGEMENT',
            icon: '/assets/icons/pulse.svg',
            command: action('clicked'),
          },
          {
            id: 'team-productivity',
            label: 'PRODUCTIVITY',
            icon: '/assets/icons/energy.svg',
            command: action('clicked'),
          },
          {
            id: 'team-wellbeing',
            label: 'WELLBEING',
            icon: '/assets/icons/calm.svg',
            command: action('clicked'),
            disabled: true,
            tooltip: {
              title: 'Team Safe Zone',
              content:
                'Montara respects your people’s privacy. These trackers are available for teams with at least 5 people.',
            },
          },
          {
            id: 'team-team-members',
            label: 'TEAM MEMBERS',
            icon: '/assets/icons/team.svg',
            command: action('clicked'),
            disabled: true,
          },
        ],
      },
      {
        label: 'ME',
        items: [
          {
            id: 'productivity',
            label: 'PRODUCTIVITY',
            icon: '/assets/icons/energy.svg',
            command: action('clicked'),
          },
          {
            id: 'wellbeing',
            label: 'WELLBEING',
            icon: '/assets/icons/calm.svg',
            command: action('clicked'),
          },
          {
            id: 'leadership',
            label: 'LEADERSHIP',
            icon: '/assets/icons/goal.svg',
            command: action('clicked'),
          },
          {
            id: 'career',
            label: 'CAREER GROWTH',
            icon: '/assets/icons/compass.svg',
            command: action('clicked'),
          },
        ],
      },
    ],
    footer: [
      {
        label: '',
        items: [
          {
            id: 'settings',
            label: 'SETTINGS',
            icon: '/assets/icons/settings.svg',
            command: action('clicked'),
          },
          {
            id: 'logout',
            label: 'LOGOUT',
            icon: '/assets/icons/log-out.svg',
          },
        ],
      },
    ],
  },
  onSelected: action('Selected'),
  onExpanded: action('Expanded'),
  defaultSelected: 'dashboard',
};

Basic.parameters = {
  jest: ['Menu.test.tsx'],
};
