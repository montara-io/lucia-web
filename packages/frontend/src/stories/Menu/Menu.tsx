import React, { useState } from 'react';
import './Menu.scss';
import Tooltip from '../Tooltip/Tooltip';
import SideNav, { NavIcon, NavItem, NavText, Subtitle } from './StyledMenu';
import Avatar from '../Avatar';
import { Tooltip as PrimeTooltip } from 'primereact/tooltip';
import classNames from 'classnames';

type ItemType = {
  id: string;
  label: string;
  icon: string;
  command?: any;
  disabled?: boolean;
  addId?: boolean;
  tooltip?: {
    title: string;
    content: string | Element;
  };
};

export type GroupOfItems = {
  label: string;
  disabled?: boolean;
  tooltip?: {
    title: string;
    content: string;
  };
  items: ItemType[];
};

export type MenuProps = {
  id: string;
  defaultSelected: string;
  model: {
    body: GroupOfItems[];
    footer: GroupOfItems[];
  };
  onExpanded?: (expanded: boolean) => void;
  profileImage: string;
};

export const Menu: React.FunctionComponent<MenuProps> = (props) => {
  const [expanded, setExpanded] = useState(true);

  const onToggle = (expanded: any) => {
    setExpanded(expanded);
    props.onExpanded && props.onExpanded(expanded);
  };

  const renderItem = (
    item: {
      id: any;
      icon: any;
      disabled: any;
      command: any;
      label: any;
      tooltip?: any;
      addId: any;
    },
    expanded: boolean,
    selected: string
  ) => {
    const { id, icon, disabled, command, label, tooltip, addId } = item;

    return (
      <>
        <NavItem
          eventKey={id}
          className={classNames(
            `NavItem-${id}`,
            selected === id
              ? 'sidenav---selected---1EK3y '
              : disabled
              ? 'disabled'
              : ''
          )}
          key={id}
          {...(addId ? { id: id } : {})}
          {...(!disabled ? { onClick: command } : {})}
        >
          <NavIcon>
            <img src={icon} alt={label} />
          </NavIcon>
          <NavText className={disabled ? 'disabled' : ''}>
            {label}
            {tooltip?.title && tooltip?.content && (
              <PrimeTooltip
                className="menu-item"
                target={`.NavItem-${id}`}
                position="right"
                autoHide={false}
              >
                <div className="title">{tooltip?.title}</div>
                <div className="content">{tooltip?.content}</div>
              </PrimeTooltip>
            )}
          </NavText>
        </NavItem>
        {!expanded && <Tooltip position="right">{label}</Tooltip>}
      </>
    );
  };

  const renderGroup = (group, expanded: boolean, selected: string) => {
    return group.map(({ items, label, disabled, tooltip }) => [
      <div
        className={`m-subtitle ${!label && 'hide'} ${!expanded && 'collapsed'}`}
        key={label}
      >
        {(expanded || (!disabled && !expanded)) && <Subtitle>{label}</Subtitle>}
      </div>,
      items.map((sub: any) => renderItem(sub, expanded, selected)),
    ]);
  };

  return (
    <div className="m-menu" data-testid="m-menu">
      <SideNav
        className={`m-sidenav${expanded ? ' expanded' : ''}`}
        onToggle={onToggle}
        expanded={expanded}
      >
        <div className={`logo${expanded ? ' expanded' : ''}`} />
        <SideNav.Toggle expanded={true}>
          <span className="small-arrow" />
        </SideNav.Toggle>
        <SideNav.Nav defaultSelected={props.defaultSelected}>
          {renderGroup(props.model.body, expanded, props.defaultSelected)}
        </SideNav.Nav>
        {props.profileImage && (
          <div className="m-profile">
            <Avatar src={props.profileImage} />
          </div>
        )}
        <SideNav.Nav
          className={`m-footer${props.profileImage ? '' : '-without-img'}`}
        >
          {renderGroup(props.model.footer, expanded, props.defaultSelected)}
        </SideNav.Nav>
      </SideNav>
    </div>
  );
};

export default Menu;
