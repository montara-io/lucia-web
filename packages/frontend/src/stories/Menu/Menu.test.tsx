import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import Menu, { MenuProps } from './Menu';

describe('Test Menu Component', () => {
  let props: MenuProps;
  const handleClick1 = jest.fn();
  const handleClick2 = jest.fn();
  const handleToggle = jest.fn();

  beforeEach(() => {
    props = {
      id: 'menu',
      profileImage: '/assets/avatar.png',
      model: {
        body: [
          {
            label: '',
            items: [
              {
                id: 'dashboard',
                label: 'DASHBOARD',
                icon: '/assets/icons/home.svg',
                command: handleClick1,
              },
            ],
          },
          {
            label: 'MY TEAM',
            items: [
              {
                id: 'team-engagement',
                label: 'ENGAGEMENT',
                icon: '/assets/icons/pulse.svg',
                command: handleClick2,
                disabled: true,
              },
              {
                id: 'team-productivity',
                label: 'PRODUCTIVITY',
                icon: '/assets/icons/energy.svg',
                command: () => {},
              },
              {
                id: 'team-wellbeing',
                label: 'WELLBEING',
                icon: '/assets/icons/calm.svg',
                command: () => {},
              },
              {
                id: 'team-team-members',
                label: 'TEAM MEMBERS',
                icon: '/assets/icons/team.svg',
                command: () => {},
              },
            ],
          },
          {
            label: 'ME',
            items: [
              {
                id: 'productivity',
                label: 'PRODUCTIVITY',
                icon: '/assets/icons/energy.svg',
                command: () => {},
              },
              {
                id: 'wellbeing',
                label: 'WELLBEING',
                icon: '/assets/icons/calm.svg',
                command: () => {},
              },
              {
                id: 'leadership',
                label: 'LEADERSHIP',
                icon: '/assets/icons/goal.svg',
                command: () => {},
              },
              {
                id: 'career',
                label: 'CAREER GROWTH',
                icon: '/assets/icons/compass.svg',
                command: () => {},
              },
            ],
          },
        ],
        footer: [
          {
            label: '',
            items: [
              {
                id: 'settings',
                label: 'SETTINGS',
                icon: '/assets/icons/settings.svg',
                command: () => {},
              },
              {
                id: 'logout',
                label: 'LOGOUT',
                icon: '/assets/icons/log-out.svg',
              },
            ],
          },
        ],
      },
      onExpanded: handleToggle,
      defaultSelected: 'dashboard',
    };
  });

  it('should fire command event', () => {
    const { getByText } = render(<Menu {...props} />);
    const menuItem = getByText('DASHBOARD');
    expect(handleClick1).toHaveBeenCalledTimes(0);
    fireEvent.click(menuItem);
    expect(handleClick1).toHaveBeenCalledTimes(1);
  });

  it('should not fire command event on disabled item', () => {
    const { getByText } = render(<Menu {...props} />);
    const menuItem = getByText('ENGAGEMENT');
    expect(handleClick2).toHaveBeenCalledTimes(0);
    fireEvent.click(menuItem);
    expect(handleClick2).toHaveBeenCalledTimes(0);
  });

  it('should fire onToggle event', () => {
    const { getByRole } = render(<Menu {...props} />);
    const toggleSwitchBtn = getByRole('button');
    expect(handleToggle).toHaveBeenCalledTimes(0);
    fireEvent.click(toggleSwitchBtn);
    expect(handleToggle).toHaveBeenCalledTimes(1);
  });

  it('Snapshot renders correctly', () => {
    const wrapper = render(<Menu {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should disable entire section', () => {
    props = {
      id: 'menu',
      profileImage: '/assets/avatar.png',
      model: {
        body: [
          {
            label: '',
            items: [
              {
                id: 'dashboard',
                label: 'DASHBOARD',
                icon: '/assets/icons/home.svg',
                command: handleClick1,
              },
            ],
          },
          {
            label: 'MY TEAM',
            disabled: true,
            items: [
              {
                id: 'team-engagement',
                label: 'ENGAGEMENT',
                icon: '/assets/icons/pulse.svg',
                command: handleClick2,
              },
              {
                id: 'team-productivity',
                label: 'PRODUCTIVITY',
                icon: '/assets/icons/energy.svg',
                command: () => {},
              },
            ],
          },
          {
            label: 'ME',
            items: [
              {
                id: 'productivity',
                label: 'PRODUCTIVITY',
                icon: '/assets/icons/energy.svg',
                command: () => {},
              },
              {
                id: 'wellbeing',
                label: 'WELLBEING',
                icon: '/assets/icons/calm.svg',
                command: () => {},
              },
              {
                id: 'leadership',
                label: 'LEADERSHIP',
                icon: '/assets/icons/goal.svg',
                command: () => {},
              },
              {
                id: 'career',
                label: 'CAREER GROWTH',
                icon: '/assets/icons/compass.svg',
                command: () => {},
              },
            ],
          },
        ],
        footer: [
          {
            label: '',
            items: [
              {
                id: 'settings',
                label: 'SETTINGS',
                icon: '/assets/icons/settings.svg',
                command: () => {},
              },
              {
                id: 'logout',
                label: 'LOGOUT',
                icon: '/assets/icons/log-out.svg',
              },
            ],
          },
        ],
      },
      onExpanded: handleToggle,
      defaultSelected: 'dashboard',
    };

    const wrapper = render(<Menu {...props} />);
    const engagementText = wrapper.getByText('ENGAGEMENT');

    expect(engagementText.parentElement?.parentElement?.outerHTML.includes('disabled')).toBe(true);

    const productivityText = wrapper.getAllByText('PRODUCTIVITY')[0];
    expect(productivityText.parentElement?.parentElement?.outerHTML.includes('disabled')).toBe(
      true,
    );

    expect(wrapper.getByTestId('m-shield')).toBeInTheDocument();
  });
});
