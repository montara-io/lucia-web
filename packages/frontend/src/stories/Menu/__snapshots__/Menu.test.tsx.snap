// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Menu Component Snapshot renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="m-menu"
        data-testid="m-menu"
      >
        <nav
          class="sc-bdnxRM jRVjHE m-sidenav expanded sidenav---sidenav---_2tBP sidenav---expanded---1KdUL"
        >
          <div
            class="logo expanded"
          />
          <button
            aria-expanded="true"
            class="sidenav---sidenav-toggle---1KRjR"
            role="button"
          >
            <span
              class="sidenav---icon-bar---u1f02"
            />
            <span
              class="sidenav---icon-bar---u1f02"
            />
            <span
              class="sidenav---icon-bar---u1f02"
            />
            <span
              class="small-arrow"
            />
          </button>
          <div
            class="sidenav---sidenav-nav---3tvij sidenav---expanded---1KdUL"
            role="menu"
          >
            <div
              class="m-subtitle hide false"
            >
              <span
                class="sc-gKAaRy dBIOPp"
              />
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-dashboard sidenav---selected---1EK3y  sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="DASHBOARD"
                    src="/assets/icons/home.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  DASHBOARD
                </div>
              </div>
            </div>
            <div
              class="m-subtitle false false"
            >
              <span
                class="sc-gKAaRy dBIOPp"
              >
                MY TEAM
              </span>
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-team-engagement disabled sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="ENGAGEMENT"
                    src="/assets/icons/pulse.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  ENGAGEMENT
                </div>
              </div>
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-team-productivity sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="PRODUCTIVITY"
                    src="/assets/icons/energy.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  PRODUCTIVITY
                </div>
              </div>
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-team-wellbeing sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="WELLBEING"
                    src="/assets/icons/calm.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  WELLBEING
                </div>
              </div>
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-team-team-members sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="TEAM MEMBERS"
                    src="/assets/icons/team.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  TEAM MEMBERS
                </div>
              </div>
            </div>
            <div
              class="m-subtitle false false"
            >
              <span
                class="sc-gKAaRy dBIOPp"
              >
                ME
              </span>
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-productivity sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="PRODUCTIVITY"
                    src="/assets/icons/energy.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  PRODUCTIVITY
                </div>
              </div>
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-wellbeing sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="WELLBEING"
                    src="/assets/icons/calm.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  WELLBEING
                </div>
              </div>
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-leadership sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="LEADERSHIP"
                    src="/assets/icons/goal.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  LEADERSHIP
                </div>
              </div>
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-career sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="CAREER GROWTH"
                    src="/assets/icons/compass.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  CAREER GROWTH
                </div>
              </div>
            </div>
          </div>
          <div
            class="m-profile"
          >
            <div
              class="p-avatar p-component p-avatar-image p-avatar-circle p-avatar-lg"
            >
              <img
                alt="avatar"
                src="/assets/avatar.png"
              />
            </div>
          </div>
          <div
            class="m-footer sidenav---sidenav-nav---3tvij sidenav---expanded---1KdUL"
            role="menu"
          >
            <div
              class="m-subtitle hide false"
            >
              <span
                class="sc-gKAaRy dBIOPp"
              />
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-settings sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="SETTINGS"
                    src="/assets/icons/settings.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  SETTINGS
                </div>
              </div>
            </div>
            <div
              class="sc-hKFxyN ljtVim NavItem-logout sidenav---sidenav-navitem---uwIJ-"
              role="presentation"
            >
              <div
                class="sidenav---navitem---9uL5T"
                role="menuitem"
                tabindex="-1"
              >
                <div
                  class="sidenav---navicon---3gCRo"
                >
                  <img
                    alt="LOGOUT"
                    src="/assets/icons/log-out.svg"
                  />
                </div>
                <div
                  class="sidenav---navtext---1AE_f"
                >
                  LOGOUT
                </div>
              </div>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="m-menu"
      data-testid="m-menu"
    >
      <nav
        class="sc-bdnxRM jRVjHE m-sidenav expanded sidenav---sidenav---_2tBP sidenav---expanded---1KdUL"
      >
        <div
          class="logo expanded"
        />
        <button
          aria-expanded="true"
          class="sidenav---sidenav-toggle---1KRjR"
          role="button"
        >
          <span
            class="sidenav---icon-bar---u1f02"
          />
          <span
            class="sidenav---icon-bar---u1f02"
          />
          <span
            class="sidenav---icon-bar---u1f02"
          />
          <span
            class="small-arrow"
          />
        </button>
        <div
          class="sidenav---sidenav-nav---3tvij sidenav---expanded---1KdUL"
          role="menu"
        >
          <div
            class="m-subtitle hide false"
          >
            <span
              class="sc-gKAaRy dBIOPp"
            />
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-dashboard sidenav---selected---1EK3y  sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="DASHBOARD"
                  src="/assets/icons/home.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                DASHBOARD
              </div>
            </div>
          </div>
          <div
            class="m-subtitle false false"
          >
            <span
              class="sc-gKAaRy dBIOPp"
            >
              MY TEAM
            </span>
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-team-engagement disabled sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="ENGAGEMENT"
                  src="/assets/icons/pulse.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                ENGAGEMENT
              </div>
            </div>
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-team-productivity sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="PRODUCTIVITY"
                  src="/assets/icons/energy.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                PRODUCTIVITY
              </div>
            </div>
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-team-wellbeing sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="WELLBEING"
                  src="/assets/icons/calm.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                WELLBEING
              </div>
            </div>
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-team-team-members sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="TEAM MEMBERS"
                  src="/assets/icons/team.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                TEAM MEMBERS
              </div>
            </div>
          </div>
          <div
            class="m-subtitle false false"
          >
            <span
              class="sc-gKAaRy dBIOPp"
            >
              ME
            </span>
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-productivity sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="PRODUCTIVITY"
                  src="/assets/icons/energy.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                PRODUCTIVITY
              </div>
            </div>
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-wellbeing sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="WELLBEING"
                  src="/assets/icons/calm.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                WELLBEING
              </div>
            </div>
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-leadership sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="LEADERSHIP"
                  src="/assets/icons/goal.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                LEADERSHIP
              </div>
            </div>
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-career sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="CAREER GROWTH"
                  src="/assets/icons/compass.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                CAREER GROWTH
              </div>
            </div>
          </div>
        </div>
        <div
          class="m-profile"
        >
          <div
            class="p-avatar p-component p-avatar-image p-avatar-circle p-avatar-lg"
          >
            <img
              alt="avatar"
              src="/assets/avatar.png"
            />
          </div>
        </div>
        <div
          class="m-footer sidenav---sidenav-nav---3tvij sidenav---expanded---1KdUL"
          role="menu"
        >
          <div
            class="m-subtitle hide false"
          >
            <span
              class="sc-gKAaRy dBIOPp"
            />
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-settings sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="SETTINGS"
                  src="/assets/icons/settings.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                SETTINGS
              </div>
            </div>
          </div>
          <div
            class="sc-hKFxyN ljtVim NavItem-logout sidenav---sidenav-navitem---uwIJ-"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <div
                class="sidenav---navicon---3gCRo"
              >
                <img
                  alt="LOGOUT"
                  src="/assets/icons/log-out.svg"
                />
              </div>
              <div
                class="sidenav---navtext---1AE_f"
              >
                LOGOUT
              </div>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
